Tests/tests/resourcestate_test.cpp

```diff
@@ -171,5 +172,17 @@ TEST(main, ResourceStateBlas) {
   rs.flush(cmd);
   }
 
+TEST(main, ResourceStateIndirect) {
+  TestCommandBuffer cmd;
+
+  ResourceState rs;
+
+  rs.onUavUsage(NonUniqResId::I_None, NonUniqResId(0x1), PipelineStage::S_Compute);
+  rs.flush(cmd);
+
+  rs.onUavUsage(NonUniqResId(0x1), NonUniqResId::I_None, PipelineStage::S_Indirect);
+  rs.flush(cmd);
```

Comment:
can you post output of this test?

Handling of indirect in `static std::string toString(ResourceAccess rs)` seem to be missing

Reply:
barrier {UavReadComp | UavWriteComp -> Indirect}